From 93cd5cf358e8fc59b9be25cdd76381cdc19ec087 Mon Sep 17 00:00:00 2001
From: "Guoqing,Zhang" <guoqingx.zhang@intel.com>
Date: Mon, 26 Jan 2015 14:24:31 +0800
Subject: [PATCH 260/269] atomisp: correct the usage of spin_lock in hmm bo
 device exit

this patch correct the wrong usage of spin_lock in hmm buffer
object device exit.

Change-Id: I5a88eadd639449b298e15c09ac23c1e48b2e8e7a
Orig-Change-Id: Ifb29af70b166ad3419924593e2d0acfe35cfae4c
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-19911
Signed-off-by: Guoqing,Zhang <guoqingx.zhang@intel.com>
Reviewed-on: https://android.intel.com:443/323788
Reviewed-on: https://icggerrit.ir.intel.com/38874
Reviewed-by: Bin B Han <bin.b.han@intel.com>
Approver: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
Build: Wei Tao <wei.tao@intel.com>
Integrator: Wei Tao <wei.tao@intel.com>
Maintainer: Wei Tao <wei.tao@intel.com>
---
 .../camera/drivers/media/pci/atomisp2/atomisp_driver/hmm/hmm_bo.c       | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/hmm/hmm_bo.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/hmm/hmm_bo.c
index a1728b8..95da2b5 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/hmm/hmm_bo.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/hmm/hmm_bo.c
@@ -560,7 +560,7 @@ void hmm_bo_device_exit(struct hmm_bo_device *bdev)
 
 		spin_lock_irqsave(&bdev->list_lock, flags);
 		list_del(&bo->list);
-		spin_lock_irqsave(&bdev->list_lock, flags);
+		spin_unlock_irqrestore(&bdev->list_lock, flags);
 
 		kmem_cache_free(bdev->bo_cache, bo);
 	}
-- 
1.9.1

